libnavigation-core/src/main/java/com/mapbox/navigation/core/history/MapboxHistoryRecorder.kt

```diff
@@ -0,0 +1,41 @@
+package com.mapbox.navigation.core.history
+
+import com.mapbox.navigation.base.options.HistoryRecorderOptions
+import com.mapbox.navigation.core.MapboxNavigation
+
+/**
+ * Interface into retrieving mapbox navigation history files. Retrieve an instance
+ * of this interface through [MapboxNavigation.historyRecorder].
+ *
+ * @see [HistoryRecorderOptions] to enable the recorder.
+ */
+interface MapboxHistoryRecorder {
+    /**
+     * The file directory where the history files are stored.
+     * Use this to purge old history files, upload later, or
+     * to use them for replay.
+     *
+     * @see [HistoryRecorderOptions] to customize this location.
+     *
+     * @return absolute path to the directory with history files.
+     *     returns null when there is no file directory
+     */
+    fun fileDirectory(): String?
+
+    /**
+     * Store history and run callback when the operation has completed.
+     * @param result Callback which run when history has been saved
```

Comment:
Let's mention that calling this method will stop the current recording, save it, and start a new one.

Reply:
i'm going to do a quick run to verify this behavior

Reply:
This is still relevant.

Reply:
Cool, not a high tech test here but here is what I did.

In the instrumentation test
```
        mapboxNavigation.registerRouteProgressObserver {
            val countDownLatch = CountDownLatch(1)
            val startTime = SystemClock.elapsedRealtime()
            mapboxNavigation.historyRecorder().saveHistory { filePath ->
                val deltaMs = (SystemClock.elapsedRealtime() - startTime)
                println("kyle_debug saved file on route progress $deltaMs ms")
                if (filePath != null) {
                    val event = MapboxHistoryReader(filePath).asSequence().toList()
                        .firstOrNull { it is HistoryEventGetStatus }
                    val elapsedTime = if (event is HistoryEventGetStatus) event.elapsedRealtimeNanos else null
                    println("kyle_debug get status time was $elapsedTime")
                }
                countDownLatch.countDown()
            }
            countDownLatch.await()
        }
```

```
2021-06-11 15:16:47.179 kyle_debug saved file on route progress 1 ms
        2021-06-11 15:16:47.180 kyle_debug get status time was 808411950719860
        2021-06-11 15:16:48.173 kyle_debug saved file on route progress 0 ms
        2021-06-11 15:16:48.173 kyle_debug get status time was 808412963599388
        2021-06-11 15:16:49.172 kyle_debug saved file on route progress 2 ms
        2021-06-11 15:16:49.174 kyle_debug get status time was 808413974559489
        2021-06-11 15:16:50.188 kyle_debug saved file on route progress 1 ms
        2021-06-11 15:16:50.188 kyle_debug get status time was 808414991876934
        2021-06-11 15:16:51.198 kyle_debug saved file on route progress 2 ms
        2021-06-11 15:16:51.200 kyle_debug get status time was 808415995656513
        2021-06-11 15:16:52.200 kyle_debug saved file on route progress 2 ms
        2021-06-11 15:16:52.202 kyle_debug get status time was 808417004319635
```

Reply:
a different run i was verifying the size, the list was always 2 (probably update location + get status)

Reply:
updated comment